src/twiliosim.cr

```diff
@@ -31,85 +36,155 @@ class Twiliosim::Server
+
+    if (body_params["error"])
+      plain_response(context, 400, body_params["error"])
+      return
+    end
+
+    from = body_params["from"]
+    to = body_params["to"]
+    verboice_url = body_params["verboice_url"]
+
+    unless from && to && verboice_url
+      plain_response(context, 500, "Internal error getting the request params")
+      return
+    end
+
+    call = TwilioCall.new(to, from, account_sid)
+    call.start()
+    @db.save_call(call)
+    response_call_created(context, call.id)
+    spawn do
+      # We give Verboice a sec to process the response. Just in case it needs it.
+      sleep 1.seconds
+
+      # verboice_url cannot be nil, so the `not_nil!` call shouldn't be here.
+      # It's included here just because of [this Crystal compiler known issue](https://github.com/crystal-lang/crystal/issues/3093)
+      handle_created_call(verboice_url.not_nil!, call)
+    end
+  end
+
+  private def get_body_params(body)
+    unless body
+      return { "error" => "Request body is missing" }
+    end
+    body = body.gets_to_end
+    unless body
+      return { "error" => "Request body is missing" }
+    end
+    if body.blank?
+      return { "error" => "Request body is missing" }
+    end
+    params = HTTP::Params.parse(body)
+    unless params.has_key?("Url")
+      return { "error" => "Body param 'Url' is missing" }
+    end
+    verboice_url = params["Url"]
+    unless params.has_key?("From")
+      return { "error" => "Body param 'From' is missing" }
+    end
+    from = params["From"]
+    unless params.has_key?("From")
+      return { "error" => "Body param 'To' is missing" }
+    end
+    to = params["To"]
+
+    { "to" => to, "from" => from, "verboice_url" => verboice_url, "error" => nil }
+  end
+
+  private def start_call(to, from, account_sid)
+    @db.start_call(to, from, account_sid)
+  end
+
+  private def response_call_created(context, sid)
+    context.response.status_code = 201
+    context.response.content_type = "application/json"
+    response = {sid: sid}
+    response.to_json(context.response)
+  end
+
+  private def handle_created_call(verboice_url : String, call : TwilioCall) : ReplyCommand | Nil
+    reply_command = call_verboice_and_reply_message(verboice_url, call, nil)
+    return unless reply_command
+    perform_response(reply_command, call)
+  end
+
+  private def call_verboice(verboice_url, call : TwilioCall, digits : Int32 | Nil) : String | Nil
+    request_params = {"AccountSid" => call.account_sid, "From" => call.from, "To" => call.to, "CallStatus" => call.status}
+    request_params["Digits"] = digits.to_s if (digits)
+    request_body = HTTP::Params.encode(request_params)
+    HTTP::Client.post(verboice_url, body: request_body) do |response|
+      empty_body_log = "Callback failed (body response is empty) - POST #{verboice_url} #{request_body} - #{response.status_code} - #{response.status_message}"
+      response_body = response.body_io.gets
```

Comment:
```suggestion
      response_body = response.body_io.gets_to_end
```

That will consume all the body (otherwise only the first line is returned), and it will never return Nil, so the following unless is no needed.